src/AnalyzerConfiguration/NI.CSharp.Analyzers.targets

```diff
@@ -4,16 +4,8 @@
     <NI_CodeAnalysisRuleSetDirectory>$(PkgNI_CSharp_Analyzers)\content</NI_CodeAnalysisRuleSetDirectory>
   </PropertyGroup>
 
-  <PropertyGroup>
-    <!-- this would more correctly be: assemblyname contains .Test. We have an issue with test utilities projects, and test.testutilities projects. -->
-    <NI_IsTestProject Condition="'$(NI_IsTestProject)' != 'True'">$(MSBuildProjectName.StartsWith("Tests."))</NI_IsTestProject>
-    <NI_IsTestProject Condition="'$(NI_IsTestProject)' != 'True'">$(MSBuildProjectName.EndsWith(".Tests"))</NI_IsTestProject>
-    <NI_IsTestProject Condition="'$(NI_IsTestProject)' != 'True'">$(MSBuildProjectName.StartsWith("!Tests."))</NI_IsTestProject>
-    <NI_IsTestUtilitiesProject Condition="'$(NI_IsTestUtilitiesProject)' != 'True'">$(MSBuildProjectName.StartsWith("TestUtilities."))</NI_IsTestUtilitiesProject>
-    <NI_IsTestUtilitiesProject Condition="'$(NI_IsTestUtilitiesProject)' != 'True'">$(MSBuildProjectName.StartsWith("!TestUtilities."))</NI_IsTestUtilitiesProject>
-    <!-- The following is needed for misnamed TestUtilities that contain xaml because the generated project for the second compile appends random characters to the end of the project name -->
-    <NI_IsTestUtilitiesProject Condition="'$(NI_IsTestUtilitiesProject)' != 'True' and '$(NI_IsTestProject)' != 'True'">$(MSBuildProjectName.Contains(".TestUtilities"))</NI_IsTestUtilitiesProject>
-
+  <!-- Rulesets are deprecated by Microsoft and we should only enable it if we're using earlier than .NET 6 -->
```

Comment:
Do you know if this is something we're required to support?

Reply:
This is a good question, but I think I would phrase it a little differently.  I wonder if it makes sense for us to freeze our .NET Framework projects on the current version or earlier, and choose to support only .NET 6+ with subsequent changes to this analyzer package.

I'd really like to be taking things out of this package more than putting more and more into it - and maybe if we froze things for .NET Framework we could do that?  For instance, we really should remove our dependencies on the microsoft analyzer packages that are supplied by the dotnet SDK - that's not needed (and causes problems) for modern .NET.

Reply:
> I wonder if it makes sense for us to freeze our .NET Framework projects on the current version or earlier, and choose to support only .NET 6+ with subsequent changes to this analyzer package.

I went in assuming we had to maintain compatibility for the .NET framework build, but if there's the option to stop supporting it I would definitely be in favor.

ASW's Everything.sln is still using 1.2.14 of this analyzer package, not even the current 2.0.11, so I think it may already be effectively frozen for all intents and purposes?  I don't know if there are other .NET framework builds in the company that want to adopt the latest analyzer changes though.

Who can make the call for stopping support?

Reply:
Good question on support. I don't know. I think someone like @nick-beer or @gregr-ni could help drive that or point us in the right direction.

Reply:
@gregr-ni - unfortunately, I think we're already in the place you describe - in ASW at least.  We have a set of .ruleset files for .NET Framework and a set of .ruleset files for .NET 6+.  Additionally, we haven't brought .NET Framework forward to the latest version of this package, so we already have issues with patches.

My thought is that this is ok - .NET Framework development is very stable at this point.  We haven't made many changes to this package that are needed by the .NET Framework build, and we haven't changed the rulesets in some time. The problem with being frozen, and the task of keeping things in sync hasn't been significant.

One upside to this is that as we're able to improve things for .NET 6+, we also have the opportunity to improve .NET Framework code that's shared with .NET 6+ code.  We'll be able to uptake new Microsoft analyzers with each .NET SDK release - and because the code is shared, both versions have the opportunity to improve.

I do think that if we want to freeze .NET Framework support, we should do that in a separate PR though - so we can focus on the task. I'm ok with letting this go forward as is.